crates/ide/src/references/rename.rs

```diff
@@ -20,29 +20,66 @@ use crate::{
     SourceChange, SourceFileEdit, TextRange, TextSize,
 };
 
+#[derive(Debug)]
+pub(crate) struct RenameError(pub(crate) String);
+
+impl RenameError {
+    pub(crate) fn new<T: Into<String>>(message: T) -> RenameError {
+        RenameError(message.into())
+    }
+}
+
+impl std::fmt::Display for RenameError {
+    fn fmt(&self, f: &mut core::fmt::Formatter<'_>) -> core::fmt::Result {
+        write!(f, "{}", self.0)
```

Comment:
I usually write this as `fmt::Display::fmt(&self.0, f)`, to makes sure that all formatter flags are properly forwarded.